Collections/script.js

```diff
@@ -1,33 +1,439 @@
+      message += " which is a fun car.";
+    } else {
+      message += " which is a boring car.";
+    }
+
+    // Wyświetlamy wiadomości
+    console.log(message);
+  });
+}
+// Wywołujemy funkcję i wyświetlamy wynik w konsoli :)
+displayUserInfo(peopleArray);
+
 // Zadanie 12.
 
+// Funkcja, która zwraca osoby mające co najmniej 40 lat i będące mężczyznami
+function getOldMen(peopleArray) {
+  // Filtrujemy osoby mające co najmniej 40 lat i będące mężczyznami
+  const oldMen = peopleArray.filter(function (person) {
+    return person.age >= 40 && person.gender === "man";
+  });
+
+  // Tworzymy wiadomość dla każdej osoby
+  const messages = oldMen.map(function (person) {
+    return (
+      "The user is " + person.name + " and he is " + person.age + " years old"
+    );
+  });
+
+  // Sumujemy wiek wszystkich znalezionych osób
+  const totalAge = oldMen.reduce(function (sum, person) {
+    return sum + person.age;
+  }, 0);
+
+  // Dodajemy wiadomość o sumie wieku
+  messages.push("Together they have " + totalAge + " years");
+
+  // zwracamu wiadomości
+  return messages;
+}
+
+const result = getOldMen(peopleArray);
+result.forEach(function (message) {
+  console.log(message);
+});
+
 // Zadanie 13.
 
+// Funkcja, która zwraca nową tablicę z posortowanymi użytkownikami od najmłodszego do najstarszego
+function sortUsersByAge(peopleArray) {
+  // Używamy metody sort do posortowania użytkowników. Używamy slice przed sortowaniem, aby utworzyć kopię oryginalnej tablicy, aby nie modyfikować jej kolejności.
+  const sortedUsers = peopleArray.slice().sort(function (a, b) {
+    // Porównujemy wiek dwóch użytkowników
+    return a.age - b.age;
+  });
+
+  // Zwracamy nową posortowaną tablicę
+  return sortedUsers;
+}
+
+const sortedUsers = sortUsersByAge(peopleArray);
+console.log(sortedUsers);
+
 // Zadanie 14.
 
+// Funkcja, która zwraca dzień tygodnia na podstawie daty urodzin
+function dayOfTheWeekYouWereBorn(birthDate) {
+  // Tworzymy obiekt Date na podstawie daty urodzin
+  const dateOfBirth = new Date(birthDate);
+
+  // Tablica z nazwami dni tygodnia
+  const daysOfWeek = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+
+  // Pobieramy dzień tygodnia z obiektu Date
+  const dayOfWeekIndex = dateOfBirth.getDay();
+
+  // Zwracamy komunikat z nazwą dnia tygodnia
+  return `You were born on ${daysOfWeek[dayOfWeekIndex]}`;
+}
+
+const result2 = dayOfTheWeekYouWereBorn("1992-06-08");
+const result3 = dayOfTheWeekYouWereBorn("2000-11-20");
+console.log(result2);
+console.log(result3);
+
 // Zadanie 15.
 
+// Funkcja, która zlicza wystąpienia każdego słowa w przekazanym ciągu znaków
+function countWordOccurrences(inputString) {
+  // Tworzymy obiekt do zapisywania ilości wystąpień słów
+  const wordCount = {};
+
+  // funkcja dzieli ciąg znaków na słowa i przekształca je na małe litery
+  const words = inputString.toLowerCase().match(/\b\w+\b/g);
```

Comment:
Komentarz się zgadza, z tym, że jest na odwrót : D Najpierw przekształcasz cały ciąg znaków do postaci małych liter, a potem wywołujesz funkcję, która zwraca listę tokenów pasujących do wyrażenia regularnego. 

BTW wystarczy tak: 

```suggestion
  const words = inputString.toLowerCase().match(/\w+/g);
```